Workspace security fixing

pysailor opened the pull request:
- Take away the View permission from Authenticated in "private" workspaces.
  This currently violates "Workspace is visible to non-members", but since we have no story to request access, we leave it like this
- Do not show buttons / links for creating content to users who do not have the permission to use them
- In the social activity stream, make sure comments are shown to all with viewer permissions, but only the commenting form is hidden 
- Add permission "Add portal content" to the workflow of workspaces to turn off inheritance
  This permission is given, in all states of the workflow, to role Contributor
- Tighten security in microblog: If the context of a StatusUpdate is a workspace, and that workspace cannot be resolved because the user does not have the permission to view it, don't silently ignore but raise an error.

pysailor committed:
We must not show the newpostbox tile to users who don't have the required permission.
But we do want to show the complete trail of status updates, including comments, just not
the form for replying to an update or comment.
FIXME: In case the newpostbox tile is not show and we immediately start showing the activity stream
we get the ugly effect that the first update is glued to the top of the WS. => Ask Cornelis
to design this special case

pysailor committed:
Take away "View" permission from authenticated for private workspaces

pysailor committed:
The workspace workflow must 1) also take away inheritance for the permission "Add portal content"
2) grant that permission to role Conributor (all states) 3) grant "Modify portal content" to role
"Editor" (all states)
if 2) and 3) are not done, the role assignments per Meta-group (Producers etc) are somewhat pointless

pysailor committed:
only show Create functions in sidebar if the user is allowed to use them

pysailor committed:
Add a third workspace with policy Consumers, and a robot test that shows that a Member cannot
create content

pysailor committed:
Add 4th workspace in setuphandlers (open), and a robot test to show that a non-member can view
published content

ghost commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/141/
Test FAILed.

pysailor committed:
fix my previous error, Editor should not have Editing permissions on the workspace itself!

ghost commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/142/
Test FAILed.

pysailor committed:
We are taking away the View permission from Authenticated on private workspaces. The test needs to be adapted accordingly

pysailor committed:
Tighten security: a uuidToObject call performs a catalog query. If the current user does not have View
permission on the referenced workspace, the lookup will return None. In this case, don't fail
silently, but raise an Attribute Error

ghost commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/143/
Test PASSed.

gyst commented on src/ploneintranet/workspace/browser/workspace.py:
```diff
@@ -26,6 +26,13 @@ def can_manage_workspace(self):
+        Does this user have the permission to add status updates
+        """
+        return api.user.has_permission(
+            "Plone Social: Add Microblog Status Update", obj=self)
```
Shouldn't this be self.context?

gyst commented on src/ploneintranet/workspace/browser/tiles/workspaces.py:
```diff
@@ -49,7 +49,7 @@ def my_workspaces(context):
             })
         except Unauthorized:
```
This whole try/except can now be dropped - that was one of the goals of this story, to be able to rely on the portal_catalog security filtering.

gyst commented on src/ploneintranet/workspace/profiles/default/workflows/ploneintranet_workspace_workflow/definition.xml:
```diff
@@ -31,6 +32,12 @@
+      <permission-role>Site Administrator</permission-role>
+      <permission-role>TeamManager</permission-role>
+      <permission-role>Contributor</permission-role>
+    </permission-map>
```
We should add owner here and in several other places but that's scheduled as a separate story Assembla 430.

gyst commented on src/ploneintranet/workspace/tests/test_workspace_workflow.py:
```diff
@@ -67,8 +70,8 @@ def test_private_workspace(self):
-        self.assertTrue(nonmember_permissions[VIEW],
-                        'Non-member cannot view private workspace')
+        self.assertFalse(nonmember_permissions[VIEW],
+                         'Non-member cannot view private workspace')
```
The error message on test fail should become "can" not "cannot".

pysailor committed:
Handle comments from @gyst

ghost commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/144/
Test PASSed.

gyst commented:
@pysailor You missed one of my comments on src/ploneintranet/workspace/browser/workspace.py

pysailor committed:
Fix one more thing, comment by @gyst

ghost commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/147/
Test PASSed.
